### PR TITLE
DOC: update 1.2.1 release notes

### PR DESCRIPTION
**Pull request by tylerjereddy**
Update the maintenance branch release notes for `1.2.1`. As usual, some manual curation beyond the automated scripts was in order, so please do check over.

### CODE REVIEW

File: doc/release/1.2.1-notes.rst [hunk 1]
```diff
@@ -8,11 +8,37 @@ SciPy 1.2.1 is a bug-fix release with no new features compared to 1.2.0.
...
 Pull requests for 1.2.1
 -----------------------
+
+* `#9612 <https://github.com/scipy/scipy/pull/9612>`__: BUG: don't use array newton unless size is greater than 1
+* `#9615 <https://github.com/scipy/scipy/pull/9615>`__: ENH: Add test for encoding
+* `#9720 <https://github.com/scipy/scipy/pull/9720>`__: BUG: stats: weighted KDE does not modify the weights array
+* `#9739 <https://github.com/scipy/scipy/pull/9739>`__: BUG: qr_updates fails if u is exactly in span Q
+* `#9725 <https://github.com/scipy/scipy/pull/9725>`__: TST: pin mingw for Azure Win CI
+* `#9736 <https://github.com/scipy/scipy/pull/9736>`__: TST: adjust to vmImage dispatch in Azure
+* `#9681 <https://github.com/scipy/scipy/pull/9681>`__: BUG: Fix failing stats tests (partial backport)
```
Comment (tylerjereddy) [inline doc/release/1.2.1-notes.rst]: Only part of the contents of this PR were included, so I added a parenthetical note.

**Comment by tylerjereddy** [comment]
I do want the CI to run through here despite doc-only change given that I'd like to do this release soon, so lack of ci skip intentional. Also, it wouldn't cover all the services anyway.

**Comment by jakevdp** [comment]
Looks good, thanks for taking care of this!

**Comment by tylerjereddy** [comment]
Windows Python 3.6 / 32 bit failed on Azure again, with details below the fold

<details>

```
2019-02-06T02:56:54.9325259Z ================================== FAILURES ===================================
2019-02-06T02:56:54.9326605Z ___________________________ TestLGMRES.test_arnoldi ___________________________
2019-02-06T02:56:54.9327561Z [gw0] win32 -- Python 3.6.4 C:\hostedtoolcache\windows\Python\3.6.4\x86\python.exe
2019-02-06T02:56:54.9327732Z 
2019-02-06T02:56:54.9328996Z self = <scipy.sparse.linalg.isolve.tests.test_lgmres.TestLGMRES object at 0x0F4D0590>
2019-02-06T02:56:54.9329116Z 
2019-02-06T02:56:54.9329281Z     @pytest.mark.skipif(python_implementation() == 'PyPy',
2019-02-06T02:56:54.9330526Z                         reason="Fails on PyPy CI runs. See #9507")
2019-02-06T02:56:54.9330731Z     def test_arnoldi(self):
2019-02-06T02:56:54.9331501Z         np.random.rand(1234)
2019-02-06T02:56:54.9331687Z     
2019-02-06T02:56:54.9331870Z >       A = eye(10000) + rand(10000, 10000, density=1e-4)
2019-02-06T02:56:54.9332045Z 
2019-02-06T02:56:54.9332941Z self       = <scipy.sparse.linalg.isolve.tests.test_lgmres.TestLGMRES object at 0x0F4D0590>
2019-02-06T02:56:54.9332996Z 
2019-02-06T02:56:54.9334086Z C:\hostedtoolcache\windows\Python\3.6.4\x86\lib\site-packages\scipy\sparse\linalg\isolve\tests\test_lgmres.py:98: 
2019-02-06T02:56:54.9335078Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
2019-02-06T02:56:54.9336167Z C:\hostedtoolcache\windows\Python\3.6.4\x86\lib\site-packages\scipy\sparse\construct.py:842: in rand
2019-02-06T02:56:54.9336592Z     return random(m, n, density, format, dtype, random_state)
2019-02-06T02:56:54.9337847Z C:\hostedtoolcache\windows\Python\3.6.4\x86\lib\site-packages\scipy\sparse\construct.py:788: in random
2019-02-06T02:56:54.9338198Z     ind = random_state.choice(mn, size=k, replace=False)
2019-02-06T02:56:54.9339194Z mtrand.pyx:1192: in mtrand.RandomState.choice
2019-02-06T02:56:54.9339256Z     ???
2019-02-06T02:56:54.9341160Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
2019-02-06T02:56:54.9341285Z 
2019-02-06T02:56:54.9341330Z >   ???
2019-02-06T02:56:54.9341498Z E   MemoryError
```

</details>


I'll manually restart that single matrix entry one more time -- if it fails again, I'll consider that a blocking issue until resolved.

We've clearly seen that test fail on PyPy before because of the skip mark, but not sure we've seen it this often on Windows 32-bit. The OpenBLAS version used in development CI could be updated both on this branch & in master to match the wheel builds, though I'm not sure that would help. I should do that tomorrow maybe.

**Comment by tylerjereddy** [comment]
Good point, I already had the first one, but I apparently forgot to backport the newer one, so I've cherry-picked gh-9617 into this PR now.

**Comment by tylerjereddy** [comment]
All green -- merging update to `1.2.1` release notes and the one missing backport.

**Pull request merged**